macro/src/lib.rs

```diff
@@ -256,36 +292,40 @@ impl FieldParser {
                     segments,
                     leading_colon: type_path.path.leading_colon,
                 };
+                let inner_mmio_path_doc_ref = inner_mmio_path
+                    .to_token_stream()
+                    .to_string()
+                    .replace(" :: ", "::");
                 self.bound_checks.push(quote! {
                     derive_mmio::is_mmio::<#inner_mmio_path>();
                 });
                 let steal_func_name = format_ident!("steal_{}", field_ident);
                 quote! {
-                    #[doc = "Obtain a reference to the inner MMIO field `"]
-                    #[doc = stringify!(#field_ident)]
-                    #[doc = "`."]
-                    #[doc = "# Lifetime"]
+                    #[doc = "Obtain a reference to the inner MMIO field "]
+                    #[doc = concat!("[", #inner_mmio_path_doc_ref, "]")]
                     #[doc = ""]
-                    #[doc = "The lifetime of the returned inner MMIO block is tied to the "]
-                    #[doc = "lifetime of this structure"]
+                    /// # Lifetime
+                    ///
+                    /// The lifetime of the returned inner MMIO block is tied to the
+                    /// lifetime of this structure
                     #[inline]
                     pub fn #field_ident(&mut self) -> #inner_mmio_path<'_> {
                         unsafe {
                             self.#steal_func_name()
                         }
                     }
 
-                    #[doc = "Obtain a reference to the inner MMIO field `"]
-                    #[doc = stringify!(#field_ident)]
-                    #[doc = "`."]
-                    #[doc = "# Lifetime and Safety"]
+                    #[doc = "Unsafely obtain a reference to the inner MMIO field `"]
```

Comment:
Unbalanced backtick here.

Also, I have a weak dislike of mising `#[doc = ...]` and `///` in the quote! block, but a fairly strong dislike for mixing them in the same comment block.

Reply:
Fair. It's an opinion rather than a hard and fast rule.